packages/xrpc/src/types.ts

```diff
@@ -188,3 +188,17 @@ export class XRPCInvalidResponseError extends XRPCError {
     )
   }
 }
+
+type IsLiteralKey<K extends PropertyKey> = {
+  string: string extends K ? 0 : 1
+  number: number extends K ? 0 : 1
+  symbol: symbol extends K ? 0 : 1
+}[K extends string ? 'string' : K extends number ? 'number' : 'symbol']
+
+type LiteralKey<T> = keyof {
+  [K in keyof T as IsLiteralKey<K> extends 1 ? K : never]: K
+}
+
+export type StrictKeyOmit<T, K extends LiteralKey<T>> = {
```

Comment:
`K extends LiteralKey<T>` is only needed for completions - that's something that you don't quite need because this is only used by a generated code. But the same filtering would have to be done elsewhere without this constraint so it feels nice to have it here